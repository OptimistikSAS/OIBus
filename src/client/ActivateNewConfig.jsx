import React from 'react'
import { Label, Button } from 'reactstrap'
import { ReactGhLikeDiff } from 'react-gh-like-diff'
import stringify from 'json-stable-stringify'
import 'react-gh-like-diff/lib/diff2html.min.css'
import apis from './services/apis'


const ActivateNewConfig = () => {
  const [configActiveJson, setConfigActiveJson] = React.useState(null)
  const [configJson, setConfigJson] = React.useState(null)

  /**
   * Acquire the Active configuration
   * @returns {void}
   */
  React.useEffect(() => {
    apis.getActiveConfig().then(({ config }) => {
      setConfigActiveJson(config)
    })
  }, [])

  /**
   * Acquire the Current Modified configuration
   * @returns {void}
   */
  React.useEffect(() => {
    apis.getConfig().then(({ config }) => {
      setConfigJson(config)
    })
  }, [])

  /**
   * Check if modified is changed compared to active
   * @param {string} active The current active config JSON string
   * @param {string} modified The current modified config JSON string
   * @returns {boolean} compare result
   */
  const compareActiveWithModified = (active, modified) => (active && modified && (active !== modified))

  /**
   * Activate new configuration
   * @returns {void}
   */
  const handleActivate = async () => {
    try {
      await apis.updateActiveConfig()
      setConfigActiveJson(configJson)
    } catch (error) {
      console.error(error)
    }
  }

  /**
   * Reset modified configuration
   * @returns {void}
   */
  const handleDecline = async () => {
    try {
      await apis.resetModifiedConfig()
      setConfigJson(configActiveJson)
    } catch (error) {
      console.error(error)
    }
  }

  // json-stable-stringify is used instead of JSON.stringify to have consistent result in alphabetical order
  const activeString = stringify(configActiveJson, { space: '  ' })
  const modifiedString = stringify(configJson, { space: '  ' })
  const isModified = compareActiveWithModified(activeString, modifiedString)

  return (
    <>
      {isModified
        ? (
          <>
            <ReactGhLikeDiff
              options={{
                originalFileName: 'Configuration',
                updatedFileName: 'Configuration',
              }}
              past={activeString}
              current={modifiedString}
            />
            <Button color="primary" onClick={() => handleActivate()}>
              Activate
            </Button>
            <Button color="primary" onClick={() => handleDecline()}>
              Decline
            </Button>
          </>
        )
        : <Label>No modifications on configuration</Label>
      }
    </>
  )
}

export default ActivateNewConfig
