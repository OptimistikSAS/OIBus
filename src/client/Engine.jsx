import React from 'react'
import Form from 'react-jsonschema-form'

// import { } from 'reactstrap'

const Engine = () => {
  const [configJson, setConfigJson] = React.useState()
  React.useEffect(() => {
    // eslint-disable-next-line consistent-return
    fetch('/config').then((response) => {
      const contentType = response.headers.get('content-type')
      if (contentType && contentType.indexOf('application/json') !== -1) {
        return response.json().then(({ config }) => {
          setConfigJson(config)
        })
      }
    })
  }, [])
  const log = type => console.log.bind(console, type)
  return (
    <>
      <h1>Engine</h1>
      <Form
        formData={configJson && configJson.engine}
        schema={Engine.schema}
        uiSchema={Engine.uiSchema}
        autocomplete="on"
        onChange={log('changed')}
        onSubmit={log('submitted')}
        onError={log('errors')}
      />
      <pre>{configJson && JSON.stringify(configJson.engine, ' ', 2)}</pre>
    </>
  )
}

export default Engine

Engine.schema = {
  title: 'Engine',
  type: 'object',
  required: ['port', 'user', 'password'],
  properties: {
    port: { type: 'number', title: 'Port', default: '2223' },
    user: { type: 'string', title: 'User', default: 'admin' },
    password: { type: 'string', title: 'Password', default: 'd74ff0ee8da3b9806b18c877dbf29bbde50b5bd8e4dad7a3a725000feb82e8f1' },
  },
}

Engine.uiSchema = {
  port: { 'ui:help': <div>the port for the web interface to fTbus</div> },
  password: { 'ui:widget': 'password' },
  adresses: { 'ui:help': <div>the list of IP addresses allowed to access the Web interface</div> },
}
