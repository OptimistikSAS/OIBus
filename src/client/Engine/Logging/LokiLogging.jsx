import React from 'react'
import PropTypes from 'prop-types'
import OIbForm from '../../components/OIbForm/OIbForm.jsx'
import { minValue, optional } from '../../../services/validation.service'

const schema = { name: 'LokiLogging' }
schema.form = {
  level: {
    type: 'OIbSelect',
    md: 3,
    options: ['none', 'silly', 'debug', 'info', 'warning', 'error'],
    defaultValue: 'info',
    help: <div>The level for the sqlite log</div>,
  },
  host: {
    type: 'OIbText',
    label: 'Host',
    md: 4,
    valid: optional(),
    defaultValue: '',
    help: <div>The host name of the loki instance</div>,
  },
  interval: {
    type: 'OIbInteger',
    newRow: false,
    md: 2,
    valid: minValue(10),
    defaultValue: 60,
    help: <div>Interval between batch of logs (in s)</div>,
  },
  username: {
    type: 'OIbText',
    defaultValue: '',
    valid: optional(),
    md: 3,
  },
  password: {
    newRow: false,
    type: 'OIbPassword',
    defaultValue: '',
    valid: optional(),
    md: 3,
  },
}

const LokiLogging = ({ logParameters, onChange }) => (
  <div>
    <h6>Loki</h6>
    <OIbForm onChange={onChange} schema={schema} name="engine.logParameters.lokiLog" values={logParameters} />
  </div>
)

LokiLogging.propTypes = {
  onChange: PropTypes.func.isRequired,
  logParameters: PropTypes.object.isRequired,
}
export default LokiLogging
