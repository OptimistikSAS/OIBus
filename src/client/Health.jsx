import React from 'react'
import { Label, Col } from 'reactstrap'
import Table from './components/table/Table.jsx'
import apis from './services/apis'
import { AlertContext } from './context/AlertContext'

const Health = () => {
  const [status, setStatus] = React.useState([])
  const { setAlert } = React.useContext(AlertContext)

  /**
   * Acquire the status
   * @returns {void}
   */
  React.useEffect(() => {
    apis.getStatus().then((response) => {
      setStatus(response)
    }).catch((error) => {
      console.error(error)
      setAlert({ text: error.message, type: 'danger' })
    })
  }, [])

  const tableRows = Object.keys(status).map((key) => (
    [
      {
        name: 'name',
        value: key,
      },
      {
        name: 'value',
        value: status[key],
      },
    ]
  ))

  return (
    <Col>
      <Label>Health status</Label>
      {tableRows && <Table headers={[]} rows={tableRows} onRowClick={() => null} />}
    </Col>
  )
}
export default Health
