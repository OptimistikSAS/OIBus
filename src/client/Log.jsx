import React from 'react'
import { FormGroup, FormText, Label, Button, Input, ListGroup, ListGroupItem, Card, CardBody, Row, Col } from 'reactstrap'
import apis from './services/apis'
import { AlertContext } from './context/AlertContext'

const Log = () => {
  const verbosityOptions = ['debug', 'info', 'warning', 'error', 'silly']
  const defaultMaxLog = 50
  const [fromDate, setFromDate] = React.useState()
  const [toDate, setToDate] = React.useState()
  const [verbosity, setVerbosity] = React.useState(verbosityOptions)
  const [logs, setLogs] = React.useState()
  const [filterText, setFilterText] = React.useState('')
  const [maxLog, setMaxLog] = React.useState(defaultMaxLog)
  const { setAlert } = React.useContext(AlertContext)

  /**
   * Handles the form's submittion and set the logs if any response
   * @returns {void}
   */
  const handleSubmit = async () => {
    try {
      const logsResponse = await apis.getLogs(fromDate, toDate, verbosity.join(','))
      // sort logs based on timestamp
      logsResponse.sort((a, b) => {
        if (a.timestamp > b.timestamp) {
          return 1
        }
        if (b.timestamp > a.timestamp) {
          return -1
        }
        return 0
      })
      setLogs(logsResponse)
      setMaxLog(defaultMaxLog)
    } catch (error) {
      console.error(error)
      setAlert({ text: error.message, type: 'danger' })
    }
  }

  /**
   * Handles the form's submittion and set the logs if any response
   * @param {string} value checked/unchecled item
   * @returns {void}
   */
  const handleVerbosityChange = (value) => {
    if (verbosity.includes(value)) {
      setVerbosity(verbosity.filter((item) => item !== value))
    } else {
      setVerbosity([...verbosity, value])
    }
  }

  /**
   * Render the filter options for logs
   * @returns {void}
   */
  const renderFilter = () => (
    <FormGroup>
      <Label for="filterText">Filter</Label>
      <Input
        type="text"
        id="filterText"
        placeholder="message contains..."
        value={filterText}
        onEnded={() => null}
        onChange={(event) => {
          setMaxLog(defaultMaxLog)
          setFilterText(event.target.value.toLowerCase())
        }}
      />
    </FormGroup>
  )

  /**
   * Render the logs
   * @returns {void}
   */
  const renderLogs = () => {
    const filteredLogs = logs.filter((item) => item.message.toLowerCase().includes(filterText))
    return (
      <Col className="log-right-panel">
        <Card>
          <Label className="label-card-title">Logs</Label>
          <CardBody className="card-body">
            <ListGroup>
              {renderFilter()}
              {filteredLogs.filter((_, index) => index < maxLog).map((item) => {
                const { id, level, message, timestamp } = item
                const date = new Date(timestamp)
                return (
                  <ListGroupItem key={id}>
                    <Label>
                      {`${date.toLocaleDateString()} ${date.toLocaleTimeString()}`}
                    </Label>
                    {` | ${level}| ${message}`}
                  </ListGroupItem>
                )
              })}
              {filteredLogs.length > maxLog ? (
                <Button color="primary" onClick={() => setMaxLog(maxLog + defaultMaxLog)}>
                  Show more...
                </Button>
              ) : null
              }
              <br />
            </ListGroup>
          </CardBody>
        </Card>
      </Col>
    )
  }

  const maxDateString = new Date().toISOString().substr(0, 16)
  return (
    <Row>
      <Col className="log-left-panel">
        <Card>
          <Label className="label-card-title">Filters</Label>
          <CardBody className="card-body">
            <FormGroup>
              <Label for="fromDate">From date</Label>
              <Input
                type="datetime-local"
                id="fromDatee"
                max={maxDateString}
                placeholder="yyyy-mm-ddThh:mm:ss+hh:mm"
                required
                onChange={(event) => {
                  const date = new Date(event.target.value).toISOString()
                  setFromDate(date)
                }}
              />
              <FormText color="muted">
                default: Current datetime - 24 hours (now - 24 hours)
              </FormText>
            </FormGroup>

            <FormGroup>
              <Label for="toDate">To date</Label>
              <Input
                type="datetime-local"
                id="toDate"
                max={maxDateString}
                placeholder="yyyy-mm-ddThh:mm:ss+hh:mm"
                required
                onChange={(event) => {
                  const date = new Date(event.target.value).toISOString()
                  setToDate(date)
                }}
              />
              <FormText color="muted">
                default: Current datetime (now)
              </FormText>
            </FormGroup>

            <FormGroup>
              <Label for="verbosity">Verbosity</Label>
              {verbosityOptions.map((item) => (
                <FormGroup check key={item}>
                  <Label check>
                    <Input
                      type="checkbox"
                      id="verbosity"
                      onChange={() => handleVerbosityChange(item)}
                      checked={verbosity.includes(item)}
                    />
                    {item}
                  </Label>
                </FormGroup>
              ))}
            </FormGroup>
            <Button color="primary" onClick={handleSubmit}>
              Show log
            </Button>
          </CardBody>
        </Card>
      </Col>
      {logs ? renderLogs() : null}
    </Row>
  )
}
export default Log
