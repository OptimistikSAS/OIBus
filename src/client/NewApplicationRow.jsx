import React from 'react'
import PropTypes from 'prop-types'
import { Button, Form, FormGroup, Label, Input } from 'reactstrap'
import Select from './components/Select.jsx'
import apis from './services/apis'
import { AlertContext } from './context/AlertContext'

const NewApplicationRow = ({ apiList, addApplication }) => {
  const [application, setApplication] = React.useState({ applicationId: '', enable: false, api: 'Console' })
  const { setAlert } = React.useContext(AlertContext)

  /**
   * Updates the application's state
   * @param {*} event The change event
   * @returns {void}
   */
  const handleChange = (event) => {
    const { target } = event
    const { value } = target
    //  update the new application's state
    setApplication((prevState) => ({ ...prevState, [target.name]: value }))
  }

  /**
   * Submits a new application
   * @returns {void}
   */
  const handleAddApplication = async () => {
    if (application.applicationId === '') return

    try {
      await apis.addNorth(application)
      // add submitted application to the table
      addApplication(application)
      // reset the line
      setApplication({ applicationId: '', enable: false, api: 'Console' })
    } catch (error) {
      console.error(error)
      setAlert({ text: error.message, type: 'danger' })
    }
  }

  return (
    <Form className="oi-add-new">
      <FormGroup>
        <Label for="Id">
          New Application ID
        </Label>
        <Input value={application.applicationId} id="Id" name="applicationId" type="text" onChange={handleChange} />
      </FormGroup>
      <FormGroup>
        <Label for="api">
          API
        </Label>
        <Select value={application.api} id="api" name="api" options={apiList} onChange={handleChange} />
      </FormGroup>
      <FormGroup>
        <Button color="primary" onClick={() => handleAddApplication()}>
          Add
        </Button>
      </FormGroup>
    </Form>
  )
}

NewApplicationRow.propTypes = {
  apiList: PropTypes.arrayOf(PropTypes.string).isRequired,
  addApplication: PropTypes.func.isRequired,
}
export default NewApplicationRow
