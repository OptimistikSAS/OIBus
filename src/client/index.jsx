import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import { Container } from 'reactstrap'

import './style/main.less'

import TopHeader from './TopHeader.jsx'
import Activation from './Activation/Activation.jsx'
import NotFound from './NotFound.jsx'
import South from './South/South.jsx'
import North from './North/North.jsx'
import Engine from './Engine/Engine.jsx'
import ConfigureApi from './North/ConfigureApi.jsx'
import ConfigureProtocol from './South/ConfigureProtocol.jsx'
import ConfigurePoints from './South/ConfigurePoints.jsx'
import Logs from './Logs/Logs.jsx'
import Health from './Health/Health.jsx'
import AlertContainer from './components/AlertContainer.jsx'
import { AlertProvider } from './context/AlertContext.jsx'
import { ConfigProvider } from './context/configContext.jsx'

const Main = () => (
  <Router>
    <>
      <ConfigProvider>
        <AlertProvider>
          <TopHeader />
          <Container className="oi-container-with-top-nav" fluid>
            <AlertContainer />
            <Switch>
              <Route exact path="/" component={Health} />
              <Route exact path="/engine" component={Engine} />
              <Route exact path="/south" component={South} />
              <Route exact path="/south/:dataSourceId" component={ConfigureProtocol} />
              <Route exact path="/south/:dataSourceId/points" component={ConfigurePoints} />
              <Route exact path="/north" component={North} />
              <Route exact path="/north/:applicationId" component={ConfigureApi} />
              <Route exact path="/log" component={Logs} />
              <Route exact path="/activation" component={Activation} />
              <Route component={NotFound} />
            </Switch>
          </Container>
        </AlertProvider>
      </ConfigProvider>
    </>
  </Router>
)

ReactDOM.render(<Main />, document.getElementById('root'))
