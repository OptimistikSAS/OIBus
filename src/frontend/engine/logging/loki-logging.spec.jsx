/**
 * @jest-environment jsdom
 */
import React from 'react'
import { act, Simulate } from 'react-dom/test-utils'

import * as ReactDOMClient from 'react-dom/client'
import { testConfig } from '../../../../tests/test-config.js'
import LokiLogging from './loki-logging.jsx'

const onChange = jest.fn()

const mockMath = Object.create(global.Math)
mockMath.random = () => 1
global.Math = mockMath

let container
let root
// eslint-disable-next-line no-undef
globalThis.IS_REACT_ACT_ENVIRONMENT = true
beforeEach(() => {
  container = document.createElement('div')
  root = ReactDOMClient.createRoot(container)
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
  root = null
})

describe('LokiLogging', () => {
  test('check LokiLogging', () => {
    act(() => {
      root.render(<LokiLogging
        logParameters={testConfig.engine.logParameters.lokiLog}
        onChange={() => (1)}
      />)
    })
    expect(container).toMatchSnapshot()
  })
  test('check change lokiLog level to "warning"', () => {
    act(() => {
      root.render(<LokiLogging
        logParameters={testConfig.engine.logParameters.lokiLog}
        onChange={onChange}
      />)
    })
    Simulate.change(document.getElementById('engine.logParameters.lokiLog.level'), {
      target: {
        value: 'warning',
        selectedIndex: 4,
      },
    })
    expect(onChange).toBeCalledWith('engine.logParameters.lokiLog.level', 'error', null, null)
    expect(container).toMatchSnapshot()
  })
  test('check change lokiLog address name to "http://test.loki.fr"', () => {
    act(() => {
      root.render(<LokiLogging
        logParameters={testConfig.engine.logParameters.lokiLog}
        onChange={onChange}
      />)
    })
    Simulate.change(document.getElementById('engine.logParameters.lokiLog.lokiAddress'), { target: { value: 'http://test.loki.fr' } })
    expect(onChange).toBeCalledWith('engine.logParameters.lokiLog.lokiAddress', 'http://test.loki.fr', null)
    expect(container).toMatchSnapshot()
  })
  test('check change lokiLog interval to 10', () => {
    act(() => {
      root.render(<LokiLogging
        logParameters={testConfig.engine.logParameters.lokiLog}
        onChange={onChange}
      />)
    })
    Simulate.change(document.getElementById('engine.logParameters.lokiLog.interval'), { target: { value: 10 } })
    expect(onChange).toBeCalledWith('engine.logParameters.lokiLog.interval', 10, null)
    expect(container).toMatchSnapshot()
  })
})
