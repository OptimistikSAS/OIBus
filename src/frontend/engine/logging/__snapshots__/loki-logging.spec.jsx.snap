// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LokiLogging check LokiLogging 1`] = `
<div>
  <div>
    <h6>
      Loki
    </h6>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.level"
          >
            Level
          </label>
          <select
            aria-invalid="false"
            class="oi-form-input form-select"
            id="engine.logParameters.lokiLog.level"
            name="engine.logParameters.lokiLog.level"
          >
            <option
              value="trace"
            >
              trace
            </option>
            <option
              value="debug"
            >
              debug
            </option>
            <option
              value="info"
            >
              info
            </option>
            <option
              value="warning"
            >
              warning
            </option>
            <option
              value="error"
            >
              error
            </option>
            <option
              value="silent"
            >
              silent
            </option>
          </select>
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The level for the Loki log
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.lokiAddress"
          >
            Host
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.lokiAddress"
            name="engine.logParameters.lokiLog.lokiAddress"
            type="text"
            value="localhost:3100"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The host name of the Loki instance
            </div>
          </small>
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.interval"
          >
            Interval (s)
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.interval"
            name="engine.logParameters.lokiLog.interval"
            type="integer"
            value="60"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              Interval between batch of logs (in s)
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.tokenAddress"
          >
            Token address
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.tokenAddress"
            name="engine.logParameters.lokiLog.tokenAddress"
            type="text"
            value=""
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The address of the token provider (need username and password)
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.username"
          >
            Username
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.username"
            name="engine.logParameters.lokiLog.username"
            type="text"
            value=""
          />
          <div
            class="invalid-feedback"
          />
        </div>
      </div>
      <div
        class="col-md-3"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.password"
          >
            Password
          </label>
          <div
            class="input-group"
          >
            <input
              aria-invalid="false"
              class="oi-form-input form-control"
              id="engine.logParameters.lokiLog.password"
              name="engine.logParameters.lokiLog.password"
              type="password"
              value=""
            />
          </div>
          <div
            class="invalid-feedback"
            style="display: inline;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LokiLogging check change lokiLog address name to "http://test.loki.fr" 1`] = `
<div>
  <div>
    <h6>
      Loki
    </h6>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.level"
          >
            Level
          </label>
          <select
            aria-invalid="false"
            class="oi-form-input form-select"
            id="engine.logParameters.lokiLog.level"
            name="engine.logParameters.lokiLog.level"
          >
            <option
              value="trace"
            >
              trace
            </option>
            <option
              value="debug"
            >
              debug
            </option>
            <option
              value="info"
            >
              info
            </option>
            <option
              value="warning"
            >
              warning
            </option>
            <option
              value="error"
            >
              error
            </option>
            <option
              value="silent"
            >
              silent
            </option>
          </select>
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The level for the Loki log
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.lokiAddress"
          >
            Host
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.lokiAddress"
            name="engine.logParameters.lokiLog.lokiAddress"
            type="text"
            value="localhost:3100"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The host name of the Loki instance
            </div>
          </small>
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.interval"
          >
            Interval (s)
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.interval"
            name="engine.logParameters.lokiLog.interval"
            type="integer"
            value="60"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              Interval between batch of logs (in s)
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.tokenAddress"
          >
            Token address
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.tokenAddress"
            name="engine.logParameters.lokiLog.tokenAddress"
            type="text"
            value=""
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The address of the token provider (need username and password)
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.username"
          >
            Username
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.username"
            name="engine.logParameters.lokiLog.username"
            type="text"
            value=""
          />
          <div
            class="invalid-feedback"
          />
        </div>
      </div>
      <div
        class="col-md-3"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.password"
          >
            Password
          </label>
          <div
            class="input-group"
          >
            <input
              aria-invalid="false"
              class="oi-form-input form-control"
              id="engine.logParameters.lokiLog.password"
              name="engine.logParameters.lokiLog.password"
              type="password"
              value=""
            />
          </div>
          <div
            class="invalid-feedback"
            style="display: inline;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LokiLogging check change lokiLog interval to 10 1`] = `
<div>
  <div>
    <h6>
      Loki
    </h6>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.level"
          >
            Level
          </label>
          <select
            aria-invalid="false"
            class="oi-form-input form-select"
            id="engine.logParameters.lokiLog.level"
            name="engine.logParameters.lokiLog.level"
          >
            <option
              value="trace"
            >
              trace
            </option>
            <option
              value="debug"
            >
              debug
            </option>
            <option
              value="info"
            >
              info
            </option>
            <option
              value="warning"
            >
              warning
            </option>
            <option
              value="error"
            >
              error
            </option>
            <option
              value="silent"
            >
              silent
            </option>
          </select>
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The level for the Loki log
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.lokiAddress"
          >
            Host
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.lokiAddress"
            name="engine.logParameters.lokiLog.lokiAddress"
            type="text"
            value="localhost:3100"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The host name of the Loki instance
            </div>
          </small>
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.interval"
          >
            Interval (s)
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.interval"
            name="engine.logParameters.lokiLog.interval"
            type="integer"
            value="60"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              Interval between batch of logs (in s)
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.tokenAddress"
          >
            Token address
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.tokenAddress"
            name="engine.logParameters.lokiLog.tokenAddress"
            type="text"
            value=""
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The address of the token provider (need username and password)
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.username"
          >
            Username
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.username"
            name="engine.logParameters.lokiLog.username"
            type="text"
            value=""
          />
          <div
            class="invalid-feedback"
          />
        </div>
      </div>
      <div
        class="col-md-3"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.password"
          >
            Password
          </label>
          <div
            class="input-group"
          >
            <input
              aria-invalid="false"
              class="oi-form-input form-control"
              id="engine.logParameters.lokiLog.password"
              name="engine.logParameters.lokiLog.password"
              type="password"
              value=""
            />
          </div>
          <div
            class="invalid-feedback"
            style="display: inline;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LokiLogging check change lokiLog level to "warning" 1`] = `
<div>
  <div>
    <h6>
      Loki
    </h6>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.level"
          >
            Level
          </label>
          <select
            aria-invalid="false"
            class="oi-form-input form-select"
            id="engine.logParameters.lokiLog.level"
            name="engine.logParameters.lokiLog.level"
          >
            <option
              value="trace"
            >
              trace
            </option>
            <option
              value="debug"
            >
              debug
            </option>
            <option
              value="info"
            >
              info
            </option>
            <option
              value="warning"
            >
              warning
            </option>
            <option
              value="error"
            >
              error
            </option>
            <option
              value="silent"
            >
              silent
            </option>
          </select>
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The level for the Loki log
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.lokiAddress"
          >
            Host
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.lokiAddress"
            name="engine.logParameters.lokiLog.lokiAddress"
            type="text"
            value="localhost:3100"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The host name of the Loki instance
            </div>
          </small>
        </div>
      </div>
      <div
        class="col-md-2"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.interval"
          >
            Interval (s)
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.interval"
            name="engine.logParameters.lokiLog.interval"
            type="integer"
            value="60"
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              Interval between batch of logs (in s)
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.tokenAddress"
          >
            Token address
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.tokenAddress"
            name="engine.logParameters.lokiLog.tokenAddress"
            type="text"
            value=""
          />
          <div
            class="invalid-feedback"
          />
          <small
            class="form-text text-muted"
          >
            <div>
              The address of the token provider (need username and password)
            </div>
          </small>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-3"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.username"
          >
            Username
          </label>
          <input
            aria-invalid="false"
            class="oi-form-input form-control"
            id="engine.logParameters.lokiLog.username"
            name="engine.logParameters.lokiLog.username"
            type="text"
            value=""
          />
          <div
            class="invalid-feedback"
          />
        </div>
      </div>
      <div
        class="col-md-3"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="engine.logParameters.lokiLog.password"
          >
            Password
          </label>
          <div
            class="input-group"
          >
            <input
              aria-invalid="false"
              class="oi-form-input form-control"
              id="engine.logParameters.lokiLog.password"
              name="engine.logParameters.lokiLog.password"
              type="password"
              value=""
            />
          </div>
          <div
            class="invalid-feedback"
            style="display: inline;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
